lib/pure/nativesockets.nim

```diff
@@ -116,6 +118,10 @@ proc `$`*(p: Port): string {.borrow.}
 proc toInt*(domain: Domain): cshort
   ## Converts the Domain enum to a platform-dependent ``cint``.
 
+proc toKnownDomain*(family: cint): Option[Domain]
```

Comment:
I don't have a strong opinion on `Option[T]`. This signature is what @Araq suggested to me (which is surprising, given that he is not a fan of Option[T] :).
I think part of the reason why `Option[T]` is not idiomatic now is because `nil` is still too idiomatic. For example, I often use `getOrDefault` on tables, then checking if return value is `nil`, which would mean the object didn't exist (assuming `nil` is not a valid value which is always the case for my tables). But for non-ref objects tables don't provide a way to get a value, checking if it exists, in single lookup (well, you can use try/except with `[]`, but it's ugly). So something like `proc get[K,V](t: Table[K,V], k: K): Option[V]` would be nice.

AF_UNKNOWN sounds a lot like AF_UNSPEC and could be confused. But maybe it is fine for this proc to return AF_UNSPEC for any unknown value (it *is* fine for `dial`).